core/src/main/java/com/avast/bytes/jdk/FileBackedBytes.java

```diff
@@ -0,0 +1,93 @@
+package com.avast.bytes.jdk;
+
+import com.avast.bytes.AbstractBytes;
+import com.avast.bytes.Bytes;
+import com.avast.bytes.internal.ByteBufferInputStream;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+/**
+ * Implementation of {@link Bytes} backed by memory mapped file  in format of {@link ByteBuffer}.
+ * <p>
+ * You create a new instance either calling {@link #from(Path)}.
+ * <p>
+ * The implementation uses {@link MappedByteBuffer} for accessing the data.
+ */
+public final class FileBackedBytes extends AbstractBytes {
+
+    private final ByteBuffer buffer;
+
+    private FileBackedBytes(final ByteBuffer buffer) {
+        this.buffer = buffer;
+    }
+
+    @Override
+    public int size() {
+        return buffer.remaining();
+    }
+
+    @Override
+    public byte byteAt(int index) {
+        return buffer.get(buffer.position() + index);
+    }
+
+    @Override
+    public byte[] toByteArray() {
+        byte[] dest = new byte[size()];
+        // create new read-only view so that we don't have to synchronize modifying the buffer's position
+        toReadOnlyByteBuffer().get(dest);
+        return dest;
+    }
+
+    @Override
+    public ByteBuffer toReadOnlyByteBuffer() {
+        return buffer.asReadOnlyBuffer();
+    }
+
+    @Override
+    public String toString(Charset charset) {
+        return getClass().getName() + "{ size = " + size() + " }";
```

Comment:
Isn't this incorrect? This is supposed to return `String` representation of the `Bytes` meaning that it should actually write out the bytes (like here https://github.com/avast/bytes/blob/master/core/src/main/java/com/avast/bytes/jdk/ByteBufferBytes.java#L54). It's not `toString` from `Object` - that does not take any parameters.